File: src/YangKai.BlogEngine.Web.Mvc/Content/js/controllers/admin/GroupController.js
var GroupController;

GroupController = [
  "$scope", "$routeParams", "Group", "Channel", function($scope, $routeParams, Group, Channel) {
    $scope.loading = true;
    Group.query({
      $filter: "IsDeleted eq false and Channel/Url eq '" + $routeParams.channel + "'"
    }, function(data) {
      $scope.list = data;
      return $scope.loading = false;
    });
    return Channel.query({
      $filter: "Url eq '" + $routeParams.channel + "'"
    }, function(data) {
      return $scope.channel = data.value[0];
    });
  }
];

File: src/YangKai.BlogEngine.Web.Mvc/Controllers/GroupController.cs
using System;
using YangKai.BlogEngine.Domain;
using YangKai.BlogEngine.Service;

namespace YangKai.BlogEngine.Web.Mvc.Controllers
{
    public class GroupController : EntityController<Group>
    {
        protected override Group CreateEntity(Group entity)
        {
            if (Proxy.Repository<Group>().Exist(p => !p.IsDeleted && p.Url == entity.Url && p.GroupId != entity.GroupId))
            {
                throw new Exception("Group has been exist.");
            }
            entity.Channel = Proxy.Repository<Channel>().Get(entity.Channel.ChannelId);
            Proxy.Repository<Group>().Add(entity);
            return entity;
        }

        protected override Group UpdateEntity(Guid key, Group update)
        {
            if (Proxy.Repository<Group>().Exist(p => !p.IsDeleted && p.Url == update.Url && p.GroupId != update.GroupId))
            {
                throw new Exception("Group has been exist.");
            }

            var entitiy = Proxy.Repository<Group>().Get(key);
            entitiy.Name = update.Name;
            entitiy.Url = update.Url;
            entitiy.Description = update.Description;
            entitiy.IsDeleted = update.IsDeleted;
            Proxy.Repository<Group>().Update(entitiy);
            return entitiy;
        }
    }
}